9/README.md

```diff
@@ -0,0 +1,322 @@
+```
+
+
+
+* Value of sequence DataType SHALL follow <sequence-value> rule:
+```abnf
+ sequence-value = *element
+```
+Where \<element\> follows specification of value of the matching \<DataType\>
+
+* Sequence DataType SHALL provide an interface to determine number of its elements.
+* Sequence DataType SHALL provide an interface to index its elements.
+
+#### Structures
+
+Value of the DevEncoded type is a structure to carry binary data with application and context specific meaning. 
+
+* Name of DevEncoded type SHALL be as follows:
+```abnf
+ DevEncoded = "DevEncoded"
+```
+
+* Value of \<DevEncoded\> SHALL be according to the \<encoded-value\> rule:
+```abnf
+ encoded-value = encoded_format encoded_data
+                 ; elements order implied by the above rule MAY be ignored
+```
+;where \<encoded_format\> has type DevString and \<encoded_data\> has type DevVarCharArray. 
+
+DevVarLongStringArray and DevVarDoubleStringArray values are structures to carry sequences of numbers 
+(integer or double precision floating point, respectively) along with sequences of text/string data.
+
+* Names DevVarLongStringArray and DevVarDoubleStringArray SHALL follow the following rules:
+```abnf
+ DevVarLongStringArray = "DevVarLongStringArray"
+ DevVarDoubleStringArray = "DevVarDoubleStringArray"
+```
+
+* Values of DevVarLongStringArray and DevVarDoubleStringArray SHALL follow rules \<long-string-value\> 
+and \<double-string-value\>, respectively:
+```abnf
+ long-string-value = lvalue svalue 
+ double-string-value = dvalue svalue
+                       ; elements order implied by the above rules MAY be ignored
+```
+;where \<lvalue\> has type DevVarLongArray, \<dvalue\> has type DevVarDoubleArray and \<svalue\> has type 
+DevVarStringArray.
+
+#### State
+
+DavState type standardize a way of describing state of Device. Its values are from predefined set of 14 names.   
+
+* DevState name SHALL be according to the following rule:
+```abnf
+ DevState = "DevState
+```
+
+* Value of DevState type SHALL follow the rule \<dev-state-value\>:
+```abnf
+ dev-state-value = "ALARM" / "INSERT" / "STANDBY" / "CLOSE" / "MOVING" / "UNKNOWN" / "DISABLE" / "OFF"
+ dev-state-value =/ "EXTRACT" / "ON" / "FAULT" / "OPEN" / "INIT" / "RUNNING"
+```
+
+#### Enumeration
+
+DevEnum data type allows to assign string Labels to DevShort values. It is valid only for Attributes. 
+
+* DevEnum name SHALL follow the following rule:
+```abnf
+ DevEnum = "DevEnum"
+```
+
+* Label values of DevEnum SHALL follow the rule \<devenum-label\>:
+```abnf
+ devenum-label = 1*VCHAR
+```  
+
+* DevEnum labels SHALL be unique in the context of an Attribute.
+* Each of DevEnum labels SHALL correspond to one and unique DevShort value. 
+* Related DevShort values SHALL be consecutive and SHALL start with 0. 
+* Values of DevEnum DataType SHALL be transported as values of DevShort.
+
+* For any of Attribute of DevEnum data type, list of Labels SHALL be available as `enum_labels` Attribute property.
```

Comment:
For any of Attribute of DevEnum data type, list of Labels SHALL be available as `enum_labels` Attribute property (cf RFC4).